File: zsh/universals/3-last/ask-openai/tests/security.py
import subprocess


def main():
    try:
        cmd = ['security', 'find-generic-password', '-s', 'openai', '-a', 'ask', '-w']
        result = subprocess.run(cmd, text=True, capture_output=True, check=True)
        print("Output:", result.stdout)
    except subprocess.CalledProcessError as e:
        print("Command failed!")
        print("Error:", e.stderr)


if __name__ == '__main__':
    main()

File: .config/hammerspoon/config/ask/security.lua
local M = {}

function M.getSecret(accountName, serviceName)
    -- TODO add try catch so I don't fubar loading my init.lua config?

    local command = string.format(
        'security find-generic-password -a %s -s %s -w 2>/dev/null',
        accountName,
        serviceName
    )

    local handle = io.popen(command)
    if not handle then
        print("Error: failed to get secret - handle is nil")
        return nil
    end
    local stdout = handle:read("*a")
    if not stdout then
        print("Error: failed to get secret - stdout is nil")
        return nil
    end
    handle:close()

    -- Trim surrounding whitespace (i.e. security output has newline)
    local trimmed = stdout:match("^%s*(.-)%s*$")
    if trimmed == "" then
        print("Error: failed to get secret - trimmed is empty")
        return nil
    end

    return trimmed
end

return M
